WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSectionHeader.cs

```diff
@@ -87,41 +92,210 @@ public class WiseSectionHeader : BaseBinaryDeserializer<SectionHeader>
             header.UnknownValue4 = data.ReadUInt32LittleEndian();
             header.FirstExecutableFileEntryLength = data.ReadUInt32LittleEndian();
             header.MsiFileEntryLength = data.ReadUInt32LittleEndian();
-            if (headerLength == 6)
-                return header;
+            // Read common values
 
-            header.UnknownValue7 = data.ReadUInt32LittleEndian();
-            header.UnknownValue8 = data.ReadUInt32LittleEndian();
-            if (headerLength == 8)
-                return header;
-
-            header.UnknownValue9 = data.ReadUInt32LittleEndian();
-            header.UnknownValue10 = data.ReadUInt32LittleEndian();
-            header.UnknownValue11 = data.ReadUInt32LittleEndian();
-            header.UnknownValue12 = data.ReadUInt32LittleEndian();
-            header.UnknownValue13 = data.ReadUInt32LittleEndian();
-            header.UnknownValue14 = data.ReadUInt32LittleEndian();
-            header.UnknownValue15 = data.ReadUInt32LittleEndian();
-            header.UnknownValue16 = data.ReadUInt32LittleEndian();
-            header.UnknownValue17 = data.ReadUInt32LittleEndian();
-            if (headerLength == 17)
-                return header;
+            if (headerLength > 6)
+            {
+                header.UnknownValue7 = data.ReadUInt32LittleEndian();
+                header.UnknownValue8 = data.ReadUInt32LittleEndian();
+            }
 
-            header.UnknownValue18 = data.ReadUInt32LittleEndian();
-            if (headerLength == 18)
-                return header;
+            if (headerLength > 8)
+            {
+                header.ThirdExecutableFileEntryLength = data.ReadUInt32LittleEndian();
+                header.UnknownValue10 = data.ReadUInt32LittleEndian();
+                header.UnknownValue11 = data.ReadUInt32LittleEndian();
+                header.UnknownValue12 = data.ReadUInt32LittleEndian();
+                header.UnknownValue13 = data.ReadUInt32LittleEndian();
+                header.UnknownValue14 = data.ReadUInt32LittleEndian();
+                header.UnknownValue15 = data.ReadUInt32LittleEndian();
+                header.UnknownValue16 = data.ReadUInt32LittleEndian();
+                header.UnknownValue17 = data.ReadUInt32LittleEndian(); 
+            }
+            
+            if (headerLength > 17)
+            {
+                header.UnknownValue18 = data.ReadUInt32LittleEndian();
+            }
+            
+            // Parse strings
+            // TODO: Count size of string section for later size verification
 
-            // TODO: Parse strings
-            header.Strings = null;
+            PreStringValuesHelper(data, header, initialOffset, localWisOffset, header.Version, out int preStringBytesSize);
+            
+            byte[][]? stringArrays = StringHelper(data, header, initialOffset, localWisOffset, preStringBytesSize);
+            if (stringArrays == null)
+            {
```

Comment:
Just a reminder that single-statement `if` can avoid having curly braces. Just make sure you have an empty newline after.

Reply:
Sorry about all these. I try to catch them, but I specifically always want curly braces when I code personally, so for PRs to other programs, I have to fight my own coding style in order to notice I'm doing it. Not an excuse, obviously, just wanted to let you know that I haven't ignored what you asked me to do, even if I will still need reminders now and then.